### PR TITLE
test: error handling for arc conformance

### PR DESCRIPTION
**Pull request by nilekhc**
**Reason for Change**:

This PR adds additional error handling while running conformance test

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: arc/conformance/plugin/conformance.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ sonobuoy-config:
   plugin-name: akv-secrets-provider-arc-conformance
   result-format: junit
 spec:
-  image: mcr.microsoft.com/oss/azure/secrets-store/provider-azure-arc-conformance:v1.0.0
+  image: stardotstar/provider-azure-arc-conformance:v1.0.1-linux-amd64
```
Comment (sozercan) [inline arc/conformance/plugin/conformance.yaml:6]: what's this image?

File: arc/conformance/plugin/arc_conformance.sh [hunk 2]
```diff
@@ -204,12 +235,17 @@ setEnviornmentVariables
 # setup keyvault
 setupKeyVault
 
+# setuo Kube config
```
Comment (sozercan) [inline arc/conformance/plugin/arc_conformance.sh:238]:
```suggestion
# setup Kube config
```

File: arc/conformance/plugin/conformance.yaml (hunk 1 above)
Comment (sozercan) [reply arc/conformance/plugin/conformance.yaml:6]: do we need `linux-amd64`?

File: arc/conformance/plugin/arc_conformance.sh [hunk 3]
```diff
@@ -48,11 +48,13 @@ setEnviornmentVariables() {
 # initialize keyvault for conformance test
 setupKeyVault() {
   # create resource group
+  echo "INFO: Creating resource group $keyvault_resource_group"
   az group create \
   --name "$keyvault_resource_group" \
   --location "$keyvault_location" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
 
   # create keyvault
+  echo "INFO: Creating KeyVault $keyvault_name"
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:57]:
```suggestion
  echo "INFO: Creating Key Vault $keyvault_name"
```

File: arc/conformance/plugin/arc_conformance.sh [hunk 4]
```diff
@@ -61,6 +63,7 @@ setupKeyVault() {
   export KEYVAULT_NAME=$keyvault_name
 
   # set access policy for keyvault
+  echo "INFO: Setting up access policies on KeyVault"
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:66]:
```suggestion
  echo "INFO: Setting up key vault access policies"
```

File: arc/conformance/plugin/arc_conformance.sh [hunk 5]
```diff
@@ -70,6 +73,7 @@ setupKeyVault() {
   --certificate-permissions get create import 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
 
   # create keyvault secret
+  echo "INFO: Creating Secret in KeyVault"
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:76]:
```suggestion
  echo "INFO: Creating secret in key vault"
```

File: arc/conformance/plugin/arc_conformance.sh [hunk 6]
```diff
@@ -79,6 +83,7 @@ setupKeyVault() {
   export SECRET_VALUE=$secret_value
 
   # create keyvault key
+  echo "INFO: Creating Keys in KeyVault"
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:86]:
```suggestion
  echo "INFO: Creating keys in key vault"
```

File: arc/conformance/plugin/arc_conformance.sh [hunk 7]
```diff
@@ -115,6 +120,7 @@ setupKeyVault() {
 
 
   # create keyvault certificate
+  echo "INFO: Creating Certs in KeyVault"
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:123]:
```suggestion
  echo "INFO: Importing certificates in key vault"
```

File: arc/conformance/plugin/arc_conformance.sh [hunk 8]
```diff
@@ -155,6 +161,32 @@ setupKeyVault() {
...
+
+  kubectl config set-cluster azure-arc-akv-test \
+    --embed-certs=true \
+    --server=${APISERVER} \
+    --certificate-authority=./ca.crt 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
+
+  kubectl config set-credentials azure-arc-akv-test --token="${TOKEN}" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
+
+  # Delete previous rolebinding if exists. And ignore the error if not found.
+  kubectl delete clusterrolebinding clusterconnect-binding || true
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:178]:
```suggestion
  kubectl delete clusterrolebinding clusterconnect-binding --ignore-not-found
```

File: arc/conformance/plugin/arc_conformance.sh [hunk 9]
```diff
@@ -155,6 +161,32 @@ setupKeyVault() {
...
+  kubectl create clusterrolebinding clusterconnect-binding --clusterrole=cluster-admin --user="${OBJECT_ID}" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py
+
+  kubectl config set-context azure-arc-akv-test \
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:181]: `azure-arc-akv-test` is used in a number of places. Maybe make it a constant and reuse it?

File: arc/conformance/plugin/arc_conformance.sh [hunk 10]
```diff
@@ -186,6 +218,11 @@ if [ -z "${ARC_CLUSTER_RG}" ]; then
   python3 /arc/setup_failure_handler.py
 fi
 
+if [ -z "${OBJECT_ID}" ]; then
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:221]: What object id is this? Could you prefix the variable name to make it more descriptive?

File: arc/conformance/plugin/arc_conformance.sh [hunk 11]
```diff
@@ -204,14 +241,20 @@ setEnviornmentVariables
 # setup keyvault
 setupKeyVault
 
+# setup Kubeconfig
+setupKubeConfig
+
 # wait for resources in ARC namespace
 waitSuccessArc="$(waitForResources deployment azure-arc)"
 if [ "${waitSuccessArc}" == false ]; then
     echo "ERROR: deployment is not avilable in namespace - azure-arc" > "${results_dir}"/error
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:250]:
```suggestion
    echo "ERROR: deployment is not available in namespace - azure-arc" > "${results_dir}"/error
```

File: arc/conformance/plugin/arc_conformance.sh (hunk 10 above)
Comment (nilekhc) [reply arc/conformance/plugin/arc_conformance.sh:221]: I have added comment for it.

File: arc/conformance/plugin/arc_conformance.sh [hunk 12]
```diff
@@ -186,6 +219,12 @@ if [ -z "${ARC_CLUSTER_RG}" ]; then
   python3 /arc/setup_failure_handler.py
 fi
 
+# OBJECT_ID is an id of the Service Principal craeted in conformance test subscription.
```
Comment (aramase) [inline arc/conformance/plugin/arc_conformance.sh:222]:
```suggestion
# OBJECT_ID is an id of the Service Principal created in conformance test subscription.
```

**Review by aramase — COMMENTED** [review]
Just one small nit! 

Feel free to ping after this [temporary change](https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/799/files#r809805473) is reverted and I can approve the PR.

File: arc/conformance/plugin/conformance.yaml (hunk 1 above)
Comment (nilekhc) [reply arc/conformance/plugin/conformance.yaml]: reverted

**Comment by nilekhc** [comment]
Hey @aramase I have reverted temporary change. PTAL when you get a change. Thanks.

**Comment by nilekhc** [comment]
> Just one small nit!
> 
> Feel free to ping after this [temporary change](https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/799/files#r809805473) is reverted and I can approve the PR.

I have reverted temporary change. PTAL when you get a change. Thanks.

**Review by aramase — APPROVED** [review]

**Pull request merged**